ceph/rados/core_workflows.py

```diff
@@ -943,6 +943,34 @@ def verify_ec_overwrites(self, **kwargs) -> bool:
         log.info(f"The image details are : {out}")
         return True
 
+    def check_compression_size(self, pool_name: str, **kwargs) -> bool:
+        """
+        Checks the given pool size against "compression_required_ratio" and verifies that data is
+        compressed in accordance to the ratio provided
+        Args:
+            pool_name: Name of the pool
+            **kwargs: additional params needed.
+                Allowed values:
+                    compression_required_ratio: ratio set on the pool for compression
+        Returns: True -> pass, False -> fail
+        """
+        log.info(f"Collecting stats about pool : {pool_name}")
+        pool_stats = self.run_ceph_command(cmd="ceph df detail")["pools"]
+        pool_1_stats = [detail for detail in pool_stats if detail["name"] == pool_name][
```

Comment:
IMO, we should break as soon as we get a match. For example
```
for detail in pool_stats:
  if detail["name"] == pool_name:
    pool_1_stats = detail["stats"
    <you could include the below checks>
    break <return True -> if the below checks are included>
else:
  raise ResourceException("no matching detail found.")
```

Reply:
made the changes to break off from the loop once the pool is found.